File: Section11_Sensor-Fusion/bumperbot_ws/src/bumperbot_localization/nodes/imu_republisher_node.py
#!/usr/bin/env python3
import rospy
from sensor_msgs.msg import Imu


def imuCallback(imu):
    imu.header.frame_id = "base_footprint_ekf"
    imu_pub.publish(imu)

if __name__== '__main__':
    rospy.init_node('imu_republisher_node')

    imu_pub = rospy.Publisher("imu_ekf", Imu, queue_size=10)
    imu_sub = rospy.Subscriber("imu", Imu, imuCallback)

    rospy.spin()

File: Section11_Sensor-Fusion/bumperbot_ws/src/bumperbot_localization/nodes/imu_republisher_node.cpp
#include <ros/ros.h>
#include <sensor_msgs/Imu.h>


ros::Publisher imu_pub;

void imuCallback(const sensor_msgs::Imu &imu)
{
    sensor_msgs::Imu new_imu;
    new_imu = imu;
    new_imu.header.frame_id = "base_footprint_ekf";
    imu_pub.publish(new_imu);
}

int main(int argc, char **argv)
{
    ros::init(argc, argv, "imu_republisher_node");
    ros::NodeHandle nh;

    imu_pub = nh.advertise<sensor_msgs::Imu>("imu_ekf", 10);
    ros::Subscriber imu_sub = nh.subscribe("imu", 1000, imuCallback);

    ros::spin();
    return 0;
}
